lib/ht_members.rb

```diff
@@ -1,17 +1,151 @@
+    "haverford" => {
+      "country_code" => "us",
+      "weight"       => 0.67,
+    }
+  })
+  puts "using canned data? #{htm.canned}"
+  puts "using mocked data? #{htm.mocked}"
+  htm.get("haverford")["country_code"]
+  htm.get("haverford")["weight"]
+
+=end
+
 class HTMembers
+  def self.template(cc, wt, sym)
+    {
+      "country_code" => cc,
+      "weight"       => wt,
+      "oclc_sym"     => sym,
+    }
+  end
 
-  def initialize(*members)
-    @members = members.empty? ? load_from_db : members
+  # If we get more cases like universityofcalifornia then this might
+  # look better as a series of SELECT ... UNION SELECT ....
+  SQL = %w<
+    SELECT DISTINCT 
+    IF(
+      mapto_inst_id='universityofcalifornia',
+      inst_id, 
+      mapto_inst_id
+    ) AS member_id,
+    country_code,
+    weight,
+    oclc_sym
+    FROM ht_institutions
+    WHERE enabled = '1'
+  >.join(' ')
+
+  CANNED_DATA = {
+    "brocku"  => HTMembers.template("ca", 0.67, "BRX"),
+    "harvard" => HTMembers.template("za", 1.33, "HUL"),
+    "umich"   => HTMembers.template("us", 1.33, "EYM"),
+    "yale"    => HTMembers.template("us", 1.33, "YUS"),
+  }
+
+  attr_reader :mocked, :canned, :members
+  def initialize(members = {})
+    @mocked  = !members.empty?
+    @canned  = false
+    @members = @mocked ? members : load_from_db
+  end
+
+  # Are all the ENV vars required for DB connection set?
+  def db_env_set?
+    problem = false
+    %w[HOST USERNAME PASSWORD PORT DATABASE].each do |name_part|
+      name = "MYSQL_#{name_part}";
+      # ENV[name] must be set, non-nil and non-empty.
+      if !ENV.key?(name) || ENV[name].nil? || ENV[name].empty? then
+        warn "ENV[#{name}] not set correctly (#{ENV[name]})"
+        problem = true
+      end
+    end
+
+    return !problem
   end
 
   def load_from_db
-    # do that
+    data = {}
+    # Attempt loading from db, use canned data if it fails.
+    begin
+      mysql_client = Mysql2::Client.new(
+        :host     => ENV["MYSQL_HOST"],
+        :username => ENV["MYSQL_USERNAME"],
+        :password => ENV["MYSQL_PASSWORD"],
+        :port     => ENV["MYSQL_PORT"],
+        :database => ENV["MYSQL_DATABASE"],
+        :connect_timeout => 5,
+      )
+
+      res = mysql_client.query(SQL)
+      res.each do |row|
+        data[row[:member_id]] = HTMembers.template(
+          row["country_code"], row["weight"], row["oclc_sym"]
+        )
+      end
+    rescue Mysql2::Error::ConnectionError => e
+      warn "#{__FILE__} is using canned data"
+      data    = CANNED_DATA
+      @canned = true
+    end
+
+    return data
   end
 
-  def get_list
-    @members
+  # Given a member_id, returns a hash of data for that member.
+  def get(member_id)
+    if @members.key?(member_id) then
+      @members[member_id]
+    else
+      warn "No member_info data for member_id:#{member_id}"
+      {"country_code"=>"xx", "weight"=>0.0, "oclc_sym"=>""}
+    end
   end
+
+end
+
+if $0 == __FILE__ then
```

Comment:
Seems like this is leftover from development?

Reply:
Or rather should be moved to the specs to ensure that we can pass in data and get back what we expect..